### PR TITLE
Feature/testing csuser csteam

### PR DESCRIPTION
**Pull request by Javier-M-V**

### CODE REVIEW

File: api/tests/csteam/test_cs_team_functions.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+from django.test import TestCase
+from api.models.csteam import CSTeam
+
+
+class CsTeamFunctionsTesCase(TestCase):
+
+    def test__str__returns_username(self):
+        #  Act
+        team_for_test = CSTeam()
+        team_for_test.name = 'cerosesenta'
```
Comment (wen96) [inline api/tests/csteam/test_cs_team_functions.py]: Better use python constructor:

```
CSTeam(name='cerosesenta')
```

File: api/tests/csuser/test_cs_user_functions.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+from api.models.csuser import CSUser
+from api.services import SteamAPIService
+
+
+class CsUserFunctionsTesCase(TestCase):
+
+    def test__str__returns_username(self):
+        #  Act
+        user_for_test = CSUser()
+        user_for_test.steam_username = 'Pepe'
```
Comment (wen96) [inline api/tests/csuser/test_cs_user_functions.py]: Better use python constructor:

```
CSUser(steam_username ='Pepe')
```

File: api/tests/csuser/test_cs_user_functions.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+        # Assert
+        self.assertEqual(str(user_for_test), user_for_test.steam_username)
+
+    def test_CSUser_init_csgo_as_none(self):
+        #  Act
+        user_for_test = CSUser()
+        # Assert
+        self.assertIsNone(user_for_test.csgo_info)
```
Comment (wen96) [inline api/tests/csuser/test_cs_user_functions.py]: What are you testing here exactly? The `python` behavior?

File: api/tests/csuser/test_cs_user_functions.py [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    def test_user_get_steam_id_returns_steam_id_when_exists(self):
+        #  Act
+        user_for_test = CSUser()
+        user_for_test.steam_id = '123'
+        result = user_for_test.get_steam_id
+        # Assert
+        self.assertEqual(result, '123')
+
+    @mock.patch.object(SteamAPIService, 'get_steam_id_from_nick_name')
+    def test_user_get_steam_id_returns_steam_id_when_does_not_exist(self, mock_get_steam_id_from_nick_name, ):
```
Comment (wen96) [inline api/tests/csuser/test_cs_user_functions.py]: better as test name something like: "test get_steam_id calls steam API when no steam_id property does not exist"?

File: api/tests/csuser/test_cs_user_functions.py [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+        # Arrange
+        mock_get_steam_id_from_nick_name.return_value = 'id'
+
+        #  Act
+        user_for_test = CSUser()
```
Comment (wen96) [inline api/tests/csuser/test_cs_user_functions.py]: I think this line should be Arrange part, shouldn't it?

File: api/tests/csuser/test_cs_user_functions.py [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+
+        mock_get_steam_info.return_value = 'steam_info'
+        mock_get_steam_id.return_value = 'Javier'
+        #  Act
+        user_for_test = CSUser()
+
+        result = user_for_test.get_steam_info
+
+        # Assert
+        self.assertEqual(result, 'steam_info')
```
Comment (wen96) [inline api/tests/csuser/test_cs_user_functions.py]: It is a good idea, that if you expect that get_steam_info and get_steam_id will be called, you assert it. You can do it with:
```
self.assertEqual(mock_get_steam_id.call_count, 1)
self.assertEqual(mock_get_steam_info.call_count, 1)
```

File: api/tests/csuser/test_cs_user_functions.py [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+        # Assert
+        self.assertEqual(result, 'id')
+
+    @mock.patch.object(SteamAPIService, 'get_steam_info')
+    @mock.patch.object(CSUser, 'get_steam_id', new_callable=mock.PropertyMock)
+    def test_get_steam_info(self, mock_get_steam_id, mock_get_steam_info):
+        # Arrange
+
+        mock_get_steam_info.return_value = 'steam_info'
+        mock_get_steam_id.return_value = 'Javier'
```
Comment (wen96) [inline api/tests/csuser/test_cs_user_functions.py]: If you don't care about the return value of get_steam_id, you can just let it at it is, you don't need a return_value, let the default value.

File: api/tests/csuser/test_cs_user_functions.py [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+        result = user_for_test.get_steam_info
+
+        # Assert
+        self.assertEqual(result, 'steam_info')
+
+    def test_cs_info_is_returned_when_exists(self):
+        #  Act
+        user_for_test = CSUser()
+        user_for_test.csgo_info = 'cs_info'
+        result = user_for_test.get_cs_info
```
Comment (wen96) [inline api/tests/csuser/test_cs_user_functions.py]: Arrange Act

File: api/tests/csuser/test_cs_user_functions.py [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+    @mock.patch.object(SteamAPIService, 'get_cs_info')
+    @mock.patch.object(CSUser, 'get_steam_id', new_callable=mock.PropertyMock)
+    def test_get_cs_info_return_empty_dic_when_get_cs_info_return_none(self, mock_get_steam_id, mock_get_cs_info):
+        # Arrange
+
+        mock_get_cs_info.return_value = []
+        mock_get_steam_id.return_value = 'Javier'
+
+        #  Act
+        user_for_test = CSUser()
```
Comment (wen96) [inline api/tests/csuser/test_cs_user_functions.py]: Arrange part?

File: api/tests/csuser/test_cs_user_functions.py [hunk 10]
```diff
@@ -0,0 +1,145 @@
...
+
+        mock_get_cs_info.return_value = [{"name": "total_kills_p90", "value": 20}]
+
+        mock_get_steam_id.return_value = 'Javier'
+
+        #  Act
+        user_for_test = CSUser()
+        result = user_for_test.get_cs_info
+
+        self.assertEqual(result['total_kills_p90'], 20)
```
Comment (wen96) [inline api/tests/csuser/test_cs_user_functions.py]: Same for this function about the Arrange and get_steam_id return_value

File: api/tests/csuser/test_cs_user_functions.py (hunk 3 above)
Comment (Javier-M-V) [reply api/tests/csuser/test_cs_user_functions.py]: Maybe something usefull... I'm testing that the constructor works and csgo_info starts as None. I did something like that in services.py.

File: api/tests/csuser/test_cs_user_functions.py (hunk 4 above)
Comment (Javier-M-V) [reply api/tests/csuser/test_cs_user_functions.py]: Absolutely agree. I'm bad giving names to functions :|

File: api/tests/csuser/test_cs_user_functions.py (hunk 5 above)
Comment (Javier-M-V) [reply api/tests/csuser/test_cs_user_functions.py]: In tests_steam_api_services.py Act and Arrange are like that.

File: api/tests/csuser/test_cs_user_functions.py (hunk 8 above)
Comment (Javier-M-V) [reply api/tests/csuser/test_cs_user_functions.py]: Maybe don't understand Use and Arrange concepts...

File: api/tests/csteam/test_cs_team_functions.py (hunk 1 above)
Comment (Javier-M-V) [reply api/tests/csteam/test_cs_team_functions.py]: New thing in python!

File: api/tests/csuser/test_cs_user_functions.py (hunk 7 above)
Comment (Javier-M-V) [reply api/tests/csuser/test_cs_user_functions.py]: Crashes with default value :/

File: api/tests/csuser/test_cs_user_functions.py (hunk 6 above)
Comment (Javier-M-V) [reply api/tests/csuser/test_cs_user_functions.py]: Interestin': new functions :D

File: api/tests/csuser/test_cs_user_functions.py (hunk 3 above)
Comment (wen96) [reply api/tests/csuser/test_cs_user_functions.py]: There are no constructor, the constructor is the default one in python, this is why I ask you what are you testing? Look at CSUser class.

**Pull request merged**